exporters/elasticsearch/include/opentelemetry/exporters/elasticsearch/es_log_exporter.h

```diff
@@ -0,0 +1,124 @@
+using json        = nlohmann::json;
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace exporter
+{
+namespace logs
+{
+/**
+ * Struct to hold Elasticsearch exporter configuration options.
+ */
+struct ElasticsearchExporterOptions
+{
+  // Configuration options to establish Elasticsearch connection
+  std::string host_;
+  int port_;
+  std::string index_;
+
+  // Maximum time to wait for response after sending request to Elasticsearch
+  int response_timeout_;
+
+  // Whether to print the status of the exporter in the console
+  bool console_debug_;
+
+  /**
+   * Constructor for the ElasticsearchExporterOptions. By default, the endpoint is
+   * localhost:9200/logs with a timeout of 30 seconds and disabled console debugging
+   * @param host The host of the Elasticsearch instance
+   * @param port The port of the Elasticsearch instance
+   * @param index The index/shard that the logs will be written to
+   * @param response_timeout The maximum time the exporter should wait after sending a request to
+   * Elasticsearch
+   * @param console_debug Print the status of the exporter methods in the console
+   */
+  ElasticsearchExporterOptions(std::string host     = "localhost",
+                               int port             = 9200,
+                               std::string index    = "logs",
+                               int response_timeout = 30,
+                               bool console_debug   = false)
+      : host_{host},
+        port_{port},
+        index_{index},
+        response_timeout_{response_timeout},
+        console_debug_{console_debug}
+  {}
+};
+
+/**
+ * The ElasticsearchLogExporter exports logs to Elasticsearch in JSON format
+ */
+class ElasticsearchLogExporter final : public sdklogs::LogExporter
+{
+public:
+  /**
+   * Create an ElasticsearchLogExporter with default exporter options.
+   */
+  ElasticsearchLogExporter();
+
+  /**
+   * Create an ElasticsearchLogExporter with user specified options.
+   * @param options An object containing the user's configuration options.
+   */
+  ElasticsearchLogExporter(const ElasticsearchExporterOptions &options);
+
+  /**
+   * Creates a recordable that stores the data in a JSON object
+   */
+  std::unique_ptr<sdk::logs::Recordable> MakeRecordable() noexcept override;
+
+  /**
+   * Exports a vector of log records to the Elasticsearch instance. Guaranteed to return after a
+   * timeout specified from the options passed from the constructor.
+   * @param records A list of log records to send to Elasticsearch.
+   */
+  sdklogs::ExportResult Export(
+      const nostd::span<std::unique_ptr<sdk::logs::Recordable>> &records) noexcept override;
+
+  /**
+   * Shutdown this exporter.
+   * @param timeout The maximum time to wait for the shutdown method to return
+   */
+  bool Shutdown(std::chrono::microseconds timeout = std::chrono::microseconds(0)) noexcept override;
+
+private:
+  // Stores if this exporter had its Shutdown() method called
+  bool isShutdown_ = false;
```

Comment:
yeah nit again: `is_shutdown_` for google naming convention

Reply:
I'm also just wondering through (relates to the ostream exporter too) - do you think this var should be initialized in the ctor instead of here?

Reply:
hmm having it here means we don't have to set it in all the different constructors (this one has 2), but I'm also not sure about if OTEL encourages this... I'll ask during the next maintainers meeting!